docs/design/mp_arktos.md

```diff
@@ -0,0 +1,172 @@
+```yaml
+spec:
+  type: mizar
+  vpcID: vpc-1a2b3c4d
+```
+The type field is the only required field, where it identifies the underline network service (which in our case is mizar). VpcID field is optional, and it lists the vpc id that is associated with this Arktos network object. If vpcID is not specified, the default VPC will be used. 
+
+The Arktos network object and VPC have **one to one relationship**, where one Arktos network object only targets to one VPC in Mizar. In other words, Arktos network object is equivalent to a vpc object in Mizar. 
+
+## Mizar VPC Object Overview: 
+In Mizar, VPC object has two required fields: ip and prefix, where in combination, they compose the CIDR of the VPC. For example, for CIDR 10.0.0.0/16, the ip = 10.0.0.0 and prefix = 16. 
+```yaml
+spec:
+  ip: 10.0.0.0
+  prefix: 16
+```
+## Proposal 
+Given the fact that Arktos network object is equivalent to vpc object in Mizar, I would like to propose two options: 
+
+### OPTION ONE: 
+Arktos adds two more optional fields into its spec, ip and prefix. This will address situations where the custom vpcID listed in Arktos network object has not been created, and Mizar needs to know the CIDR range in order to create a VPC object. The spec should show something like this: 
+```yaml
+spec:
+  type: mizar
+  vpcID: vpc-1a2b3c4d
+  ip: 10.0.0.0
+  prefix: 16
+```
+When new VPC is created and mapped with Arktos network object, the data is stored as a key/value pair in Mizar’s object store, ```self.store.vpcs_arktosnet_store[arktosnet.name][vpc.name] = vpc_object```
+
+### OPTION TWO: 
+Arktos can directly use Mizar’s VPC CRD to define its network object. As mentioned in Arktos network object overview section, Arktos network object is equivalent to VPC object in Mizar. Thus, Arktos can adapt Mizar’s VPC CRD to define Arktos network object: 
+```yaml
+apiVersion: mizar.com/v1
+kind: Vpc
+metadata:
+  name: vpc0
+spec:
+  ip: 10.0.0.0
+  prefix: 16
+  status: Init
+  dividers: 2
+```
+where status is always “Init” when creating a new VPC object, and dividers specifies the number of dividers/subnets desired. The ip and prefix fields as mentioned previous, together they define the CIDR range of the VPC.
+
+### Network Object Association
+
+In both options, the association of all network objects are stored as key/value sets in Mizar’s object store. For instance, 
+
+The Arktos network object is associated with Mizar vpc as: 
+	```self.store.vpcs_arktosnet_store[arktosnet.name] = {}```
+    ```self.store.vpcs_arktosnet_store[arktosnet.name][vpc.name] = vpc_object```
+
+The Arktos network object is associated with services in similar ways: 
+    ```self.store.services_arktosnet_store[arktosnet.name] = {}```
+    ```self.store.services_arktosnet_store[arktosnet.name][service.name] = service_object```
+
+The subnet object is associated with pods as: 
+    ```self.store.pods_net_store[net.name] = {}```
+    ```self.store.pods_net_store[net.name][pod.name] = pod_object```
```

Comment:
Do we need a new "pod" object here?
Can't we just use the preexisting endpoint object?

Reply:
Correct, pod = ep. Not need to have pods_net_store. It should be our existing eps_net_store